src/IssueViz/Models/AnalysisIssueLocationVisualization.cs

```diff
@@ -0,0 +1,47 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarLint.VisualStudio.Core.Analysis;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Models
+{
+    internal interface IAnalysisIssueLocationVisualization
```

Comment:
In the prototype we also included LineHash and SourceText but it doesn't feel right to have them here. 
IsNavigable is calculated by a service, `ILocationNavigationChecker`, so I think that should be the place to calculate the hash and compare.

Reply:
Does that mean the line text data and the hashing calculation would be repeated each time navigability is checked?
How is the navigation checker going to fetch the text (which should be the text in the document at the point the issue was analysed)?

Reply:
As discussed offline, LineHash might be part of IAnalysisIssue and IAnalysisIssueLocation. In which case, no further changes would be required to the Visualization objects (as they hold a reference to the original data objects).